frontend/frontend-filters.php

```diff
@@ -121,7 +121,15 @@ public function getarchives_join( $sql, $r ) {
 	 * @return string modified WHERE clause
 	 */
 	public function getarchives_where( $sql, $r ) {
-		return ! empty( $r['post_type'] ) && $this->model->is_translated_post_type( $r['post_type'] ) ? $sql . $this->model->post->where_clause( $this->curlang ) : $sql;
+		if ( ! $this->curlang instanceof PLL_Language || empty( $r['post_type'] ) ) {
+			return $sql;
+		}
+
+		if ( ! $this->model->is_translated_post_type( $r['post_type'] ) ) {
+			return $sql;
+		}
```

Comment:
Suggest to handle one variable per test.
```suggestion
		if ( ! $this->curlang instanceof PLL_Language ) ) {
			return $sql;
		}

		if ( empty( $r['post_type'] || ! $this->model->is_translated_post_type( $r['post_type'] ) ) {
			return $sql;
		}
```